NFIQ2/NFIQ2Algorithm/src/tool/nfiq2_ui_refresh.cpp

```diff
@@ -859,18 +837,33 @@ main(int argc, char **argv)
 	double timeInit = 0.0;
 	timerInit.startTimer();
 
-	std::string modelFile;
-	std::string hash;
+	std::shared_ptr<NFIQ::ModelInfo> modelInfoObj {};
 
 	try {
-		std::tie(modelFile, hash) = NFIQ2UI::parseModel(arguments);
+		modelInfoObj = NFIQ2UI::parseModel(arguments);
 	} catch (const NFIQ2UI::Exception &e) {
 		std::cerr << "Unable to extract model information. " << e.what()
 			  << "\n";
 		return EXIT_FAILURE;
 	}
 
-	const NFIQ::NFIQ2Algorithm model(modelFile, hash);
+	logger->debugMsg("Model Name: " + modelInfoObj->getModelName());
+	logger->debugMsg("Model Trainer: " + modelInfoObj->getModelTrainer());
+	logger->debugMsg(
+	    "Model Description: " + modelInfoObj->getModelDescription());
+	logger->debugMsg("Model Version: " + modelInfoObj->getModelVersion());
+	logger->debugMsg("Model Path: " + modelInfoObj->getModelPath());
+	logger->debugMsg("Model Hash: " + modelInfoObj->getModelHash());
+
+	std::shared_ptr<NFIQ::NFIQ2Algorithm> model {};
+	try {
+		model = std::make_shared<NFIQ::NFIQ2Algorithm>(
+		    modelInfoObj->getModelPath(), modelInfoObj->getModelHash());
```

Comment:
Suggest adding an `NFIQ2Algorithm` constructor that takes a `const ModelInfo&` and forwards the information to this constructor as a start.